### DISCUSSION TITLE
large from reset API is slow

### DISCUSSION THREAD

**Discussion by silverWolf818** [react-hook-form/react-hook-form#8348] [Q&A]
large from reset API is slow.

The way I use it is as follows:

```
reset(undefined, {
    keepValues: true,
  })
```

<img width="1766" alt="image" src="https://user-images.githubusercontent.com/7360502/168025624-f9f4105b-4588-429c-9a28-ffe074158858.png">

use 4000ms to reset form is very slow.

so I need API only reset form state like keepValues, keepErrors, keepDirty

**Comment by bluebill1049** [comment]
We wouldn't accept a feature request as such with very little context or purpose solution.  please provide more context, or a codesandbox and etc, how many components that you have mounted on the screen.

**Reply by silverWolf818** [reply]
about 1000 components useController in screen

**Reply by silverWolf818** [reply]
I'll provide an example later

**Reply by bluebill1049** [reply]
I think that explains, do all 1000 need to be mounted in the view? have you consider virtual inputs?

**Comment by eraffel-MDSol** [comment]
I'm having a similar problem. My table has 2,000 rows, each row with three <Controller /> components. Resets are incredibly slow. Anything I should be doing, or is virtualization the only good option here?

**Reply by marcebdev** [reply]
I'm not sure how virtualization would assist you in terms of form state specifically, unless you're saying your table is unpaginated and RHF is likely mounting inputs for 2k rows.

But why are you resetting, resetting does pretty much force reinitialize a bunch of RHF logic? Are you using it correctly?

**Reply by eraffel-MDSol** [reply]
the table is unpaginated. We're resetting if the user wants to abort their changes and return the state back to the original unmodified state
